apps/validator/src/main/scala/org/lfdecentralizedtrust/splice/validator/store/db/DbValidatorInternalStore.scala

```diff
@@ -64,14 +64,14 @@ class DbValidatorInternalStore(
       .storeId
       .getOrElse(throw new RuntimeException("Using storeId before it was assigned"))
 
-  def initializeState()(implicit tc: TraceContext): Future[Unit] = {
-    for {
-      result <- StoreDescriptorStore
-        .getStoreIdForDescriptor(storeDescriptor, storage)
-    } yield {
-      internalState.updateAndGet(_.copy(storeId = Some(result)))
-      ()
-    }
+  def initialize()(implicit tc: TraceContext): Future[DbValidatorInternalStore] = {
```

Comment:
Why do we need to do this? I would much rather keep it explicit returning unit as it much better reflects the side effects it has. 
Though as a side note why can't we have this done as part of the constructor and we need the side effect?